include/fmt/color.h

```diff
@@ -270,54 +424,34 @@ inline void reset_color<wchar_t>(FILE *stream) FMT_NOEXCEPT {
 
 template <
   typename S, typename Char = typename internal::char_t<S>::type>
-void vprint_rgb(rgb fd, const S &format,
-                basic_format_args<typename buffer_context<Char>::type> args) {
-  internal::fputs<Char>(internal::make_foreground_color<Char>(fd), stdout);
-  vprint(format, args);
-  internal::reset_color<Char>(stdout);
-}
-
-template <
-  typename S, typename Char = typename internal::char_t<S>::type>
-void vprint_rgb(rgb fd, rgb bg, const S &format,
+void vprint_text_format(const text_format &tf, const S &format,
                 basic_format_args<typename buffer_context<Char>::type> args) {
-  internal::fputs<Char>(internal::make_foreground_color<Char>(fd), stdout);
-  internal::fputs<Char>(internal::make_background_color<Char>(bg), stdout);
+  internal::fputs<Char>(internal::make_emphasis<Char>(tf.get_emphasis()), stdout);
+  if (tf.has_foreground())
+    internal::fputs<Char>(
+        internal::make_foreground_color<Char>(tf.get_foreground()), stdout);
+  if (tf.has_background())
+    internal::fputs<Char>(
+        internal::make_background_color<Char>(tf.get_background()), stdout);
   vprint(format, args);
   internal::reset_color<Char>(stdout);
 }
 
 /**
   Formats a string and prints it to stdout using ANSI escape sequences to
-  specify foreground color 'fd'.
-  Example:
-    fmt::print(fmt::color::red, "Elapsed time: {0:.2f} seconds", 1.23);
- */
-template <typename String, typename... Args>
-typename std::enable_if<internal::is_string<String>::value>::type
-print(rgb fd, const String &format_str, const Args & ... args) {
-  internal::check_format_string<Args...>(format_str);
-  typedef typename internal::char_t<String>::type char_t;
-  typedef typename buffer_context<char_t>::type context_t;
-  format_arg_store<context_t, Args...> as{args...};
-  vprint_rgb(fd, format_str, basic_format_args<context_t>(as));
-}
-
-/**
-  Formats a string and prints it to stdout using ANSI escape sequences to
-  specify foreground color 'fd' and background color 'bg'.
+  specify text formatting.
   Example:
-    fmt::print(fmt::color::red, fmt::color::black,
+    fmt::print(fmt::emphasis::bold | fmt::color::red,
```

Comment:
Do we need to wrap `fmt::color::red` in `fg`?

Reply:
Yes of course. :)